### PR TITLE
#169257388 Set up Unit Testing

### PR DESCRIPTION
**Pull request by gildniy**
#### What does this PR do?
Configures the app to Unit Testing

#### Description of Task to be completed
- [x] Add `mocha`, `chai` and `chai-http`
- [x] Create `tests` folder at the root of the project to hold all test files
- [x] Update the `test` script in package.json to run mocha in the `tests` folder

#### How should this be manually tested?
- Clone the repo and run install dependencies
- Run `npm test` and in the terminal, a message will be printed showing whether your tests are passing or not

#### Any background context you want to provide?
To ensure the back end has all functionality working as intended. You may use Mocha as the test runner and `chai` and `chai-http` for the unit tests.

#### What are the relevant pivotal tracker stories?
[#169257388](https://www.pivotaltracker.com/story/show/169257388)

#### Screenshots (if appropriate)
N/A

### CODE REVIEW

**Review by nshaphan — DISMISSED** [review]

File: .gitignore [hunk 1]
```diff
@@ -92,4 +92,10 @@ typings/
 # DynamoDB Local files
 .dynamodb/
 
+#Ide files
+.idea
+
+#Lock files
+package-lock.json
```
Comment (RaymondMwaura) [inline .gitignore]: I think you should not add the 'package-lock.json' file to '.gitignore'.

**Review by RaymondMwaura — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (gildniy) [reply .gitignore]: Yeah you right let me remove it

**Comment by timolinn** [comment]
@nshaphan @gildniy @Mastel22 @RaymondMwaura @b0nbon1 Mocha vs Jest ❓

**Comment by RaymondMwaura** [comment]
> @nshaphan @gildniy @Mastel22 @RaymondMwaura @b0nbon1 Mocha vs Jest ❓

I propose Mocha.

**Review by RaymondMwaura — DISMISSED** [review]

**Comment by b0nbon1** [comment]
> @nshaphan @gildniy @Mastel22 @RaymondMwaura @b0nbon1 Mocha vs Jest ❓

Me too I propose mocha

**Comment by nshaphan** [comment]
> @nshaphan @gildniy @Mastel22 @RaymondMwaura @b0nbon1 Mocha vs Jest ❓

I prefer mocha

**Review by timolinn — DISMISSED** [review]

**Review by b0nbon1 — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -4,7 +4,7 @@
   "description": "A Social platform for the creative at heart",
   "main": "index.js",
   "scripts": {
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "test": "mocha --require @babel/register src/tests/*.js"
```
Comment (b0nbon1) [inline package.json]: I think before the scripts start executing you should change the environment, may add NODE_ENV=test. so as during testing it changes the environment

**Review by b0nbon1 — CHANGES_REQUESTED** [review]

**Review by nshaphan — DISMISSED** [review]

File: package.json (hunk 2 above)
Comment (timolinn) [reply package.json]: @gildniy

Comment (gildniy) [reply package.json]: You're right! Thanks

Comment (gildniy) [reply package.json]: > I think before the scripts start executing you should change the environment, may add NODE_ENV=test. so as during testing it changes the environment

You're right, I implemented that.

**Review by b0nbon1 — DISMISSED** [review]

**Review by nshaphan — DISMISSED** [review]
Looks great

**Review by RaymondMwaura — DISMISSED** [review]
You made the changes we suggested and it works. Good job!

**Review by timolinn — APPROVED** [review]

**Review by b0nbon1 — APPROVED** [review]

**Review by nshaphan — APPROVED** [review]

**Pull request merged**